### PR TITLE
Fix: Code editor width is not being set properly when open with readme, Land on cluster detail by default, Warning card missing in Node detail 

### PR DESCRIPTION
**Pull request by Aravind-2023**
# Description
1.Code editor width is not being set properly when open with readme in BaseDeployment Page, 
2.Land on cluster detail by default for clusters section, 
3.Warning card missing in Node detail in clusters after even applying taints,
4.Env dropdown alphabetical sorting,Sort environments alphabetically,Make env dropdown searchable,Env dropdown alphabetical sorting,
5.Logo position of devtron
6.Pagination should persist when we referesh in clusters section
7.In Deployment History,we were getting undefined environment even if we select some specific environment in case of no deployments.

Screenshot 2023-05-25 at 1 02 03 PM
Screenshot 2023-05-25 at 1 03 45 PM

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/deploymentConfig/DeploymentTemplateView.tsx [hunk 1]
```diff
@@ -771,18 +771,8 @@ export const DeploymentTemplateEditorView = ({
...
-                <div className="dt-readme dc__border-right">
-                    <div className="code-editor__header flex left fs-12 fw-6 cn-9">Readme</div>
-                    {chartConfigLoading ? (
-                        <Progressing pageLoader />
-                    ) : (
-                        <MarkDown markdown={readme} className="dt-readme-markdown" />
-                    )}
-                </div>
-            )}
+    const displayCodeEditor = (): JSX.Element => {
```
Comment (shivani170) [inline src/components/deploymentConfig/DeploymentTemplateView.tsx:774]: Please rename it to renderCodeEditor or something related with the prefix render while rendering JSX element

File: src/components/app/details/cicdHistory/Sidebar.tsx [hunk 2]
```diff
@@ -76,7 +76,8 @@ const Sidebar = React.memo(({ type, filterOptions, triggerHistory, hasMore, setP
             return HISTORY_LABEL.ENVIRONMENT
         }
     }
-
+    filterOptions.sort((a,b)=>(a.label > b.label)?1:-1)
```
Comment (shivani170) [inline src/components/app/details/cicdHistory/Sidebar.tsx:79]: What if filteredOption doesn't have any value? Is it handled for the empty case ? otherwise sorting in filterOptions will break

File: src/components/app/details/cdDetails/CDDetails.tsx [hunk 3]
```diff
@@ -99,6 +99,12 @@ export default function CDDetails() {
                     ?.deploymentAppType,
             )
         }
+        if(result && result[0]){
+            let _selectedEnvironment=  (result[0]['value']?.result || []).find((envData) => {
+                return +envId===envData.environmentId
```
Comment (shivani170) [inline src/components/app/details/cdDetails/CDDetails.tsx:104]: Please format the selection and combine the result checks for 96 & 102 together.

File: src/components/app/Overview/Overview.tsx [hunk 4]
```diff
@@ -282,7 +282,9 @@ export default function AppOverview({ appMetaInfo, getAppMetaInfoRes, isJobOverv
     }
 
     const renderDeploymentComponent = () => {
+
         if (otherEnvsResult?.[0]?.result?.length > 0) {
+            otherEnvsResult[0].result.sort((a,b)=>(a.environmentName>b.environmentName)?1:-1)
```
Comment (shivani170) [inline src/components/app/Overview/Overview.tsx:287]: Format the selection

File: src/components/ClusterNodes/ClusterNodeContainer.tsx [hunk 5]
```diff
@@ -45,10 +46,10 @@ export default function ClusterNodeContainer() {
     return (
         <Switch>
             <Route path={URLS.CLUSTER_LIST} exact>
-                <ClusterList imageList={imageList} isSuperAdmin={isSuperAdmin} namespaceList={namespaceDefaultList} />
+                <ClusterList imageList={imageList} isSuperAdmin={isSuperAdmin} namespaceList={namespaceDefaultList}  />
             </Route>
-            <Route path={`${URLS.CLUSTER_LIST}/:clusterId`} exact>
-                <NodeList imageList={imageList} isSuperAdmin={isSuperAdmin} namespaceList={namespaceDefaultList} />
+            <Route path={`${URLS.CLUSTER_LIST}/:clusterId`}exact>
+                <NodeList imageList={imageList} isSuperAdmin={isSuperAdmin} namespaceList={namespaceDefaultList}   />
```
Comment (shivani170) [inline src/components/ClusterNodes/ClusterNodeContainer.tsx:52]: Cant see any changes here. If not then please revert the changes in this file

File: src/components/ClusterNodes/ClusterDetails.tsx [hunk 6]
```diff
@@ -621,7 +631,7 @@ export default function ClusterDetails({ imageList, isSuperAdmin, namespaceList,
             return '-'
         }
     }
-
+console.log(match.url)
```
Comment (shivani170) [inline src/components/ClusterNodes/ClusterDetails.tsx:634]: remove it

**Review by shivani170 — APPROVED** [review]

File: src/components/app/details/cicdHistory/Sidebar.tsx [hunk 7]
```diff
@@ -76,7 +76,8 @@ const Sidebar = React.memo(({ type, filterOptions, triggerHistory, hasMore, setP
             return HISTORY_LABEL.ENVIRONMENT
         }
     }
-
+    filterOptions?.sort((a,b)=>(a.label > b.label)?1:-1)
```
Comment (vivek-devtron) [inline src/components/app/details/cicdHistory/Sidebar.tsx:79]: We should put this before the _filterOptions on line no 68 so that we need to do the sorting only one time

File: src/components/common/navigation/navigation.scss [hunk 8]
```diff
@@ -160,6 +160,7 @@
         height: auto;
         opacity: 1;
         fill: #ffffff;
+        padding-top: 20px;
```
Comment (vivek-devtron) [inline src/components/common/navigation/navigation.scss:163]: We should use pt-20 class instead of this

File: src/components/ClusterNodes/ClusterDetails.tsx [hunk 9]
```diff
@@ -81,6 +85,12 @@ export default function ClusterDetails({ imageList, isSuperAdmin, namespaceList,
             setFixedNodeNameColumn(windowWidth < clientWidth || windowWidth < appliedColumnDerivedWidth)
         }
     }, [appliedColumns])
+  
+    useEffect(() => {
+        const qs=queryString.parse(location.search)
```
Comment (vivek-devtron) [inline src/components/ClusterNodes/ClusterDetails.tsx:90]: We should put a failsafe check for offset

**Review by vivek-devtron — APPROVED** [review]
